gui/seedwatch.lua

```diff
@@ -0,0 +1,310 @@
+-- config ui for seedwatch
+
+local gui = require('gui')
+local widgets = require('gui.widgets')
+local plugin = require('plugins.seedwatch')
+
+local PROPERTIES_HEADER = '   Quantity     Target   '
+local REFRESH_MS = 10000
+local MAX_TARGET = 2147483647
+--
+-- SeedSettings
+--
+SeedSettings = defclass(SeedSettings, widgets.Window)
+SeedSettings.ATTRS{
+    lockable=false,
+    frame={l=5, t=5, w=35, h=9},
+    data={id='', name='', quantity=0, target=0,},
```

Comment:
it is dangerous to put mutable tables as default values in ATTRS since all instances will share the same state. Since this is not actually set by the parent view, you'd be better off initializing the `data` field in `SeedSettings:init()`.

Now, it's not *really* an issue here since SeedSettings is effectively a singleton, but I'd rather this antipattern not appear anywhere in our code for other script writers to copy.

Reply:
these attr are set each time in refresh_data, so removed.